### PR TITLE
Remove container root dir from an aborted start

### PR DESCRIPTION
**Pull request by crosbymichael**
If runc was SIGKILL'd or something happened and the container was not
able to start and runc died as well then we could get into the state
where `$root/<containerid>` exists but `$root/<containerid>/state.json`
does not.  This will not allow libcontainer to load the container to
call the delete function as it has no data on the container other than
its id.  We should just remove it in runc so that that system matches
what runc sees for the container.

Signed-off-by: Michael Crosby <crosbymichael@gmail.com>

### CODE REVIEW

File: delete.go [hunk 1]
```diff
@@ -17,6 +23,15 @@ status of "ubuntu01" as "destroyed" the following will delete resources held for
 	Action: func(context *cli.Context) {
 		container, err := getContainer(context)
 		if err != nil {
+			if lerr, ok := err.(libcontainer.Error); ok && lerr.Code() == libcontainer.ContainerNotExists {
+				// if there was an aborted start or something of the sort then the container's directory could exist but
+				// libcontainer does not see it because the state.json file inside that directory was never created.
+				path := filepath.Join(context.GlobalString("root"), context.Args().First())
+				if _, err := os.Stat(path); err == nil {
+					os.RemoveAll(path)
```
Comment (dqminh) [inline delete.go]: RemoveAll also does `stat` so i dont think we need to check here.

Comment (crosbymichael) [inline delete.go]: I thought about that but I really didnt want to call removeAll before making sure that this existed.   What do you think?

Comment (cyphar) [inline delete.go]: I think it's fine, since according to the docs:

> [...] If the path does not exist, RemoveAll returns nil (no error).

**Comment by mrunalp** [comment]
LGTM

**Comment by hqhq** [comment]
LGTM

**Pull request merged**